network-config-analyzer/GenericYamlParser.py

```diff
@@ -186,3 +201,94 @@ def _get_connection_set_from_properties(dest_ports, method_set=MethodSet(True),
         res = ConnectionSet()
         res.add_connections('TCP', tcp_properties)
         return res
+
+
+from PeerContainer import PeerContainer  # noqa: E402
+
+
+class IstioGenericYamlParser(GenericYamlParser):
+    """
+    A class for istio yaml parser , common methods for istio policies parsers
+    """
+
+    def __init__(self, policy, peer_container, file_name=''):
+        """
+        :param dict policy: The istio policy object as provided by the yaml parser
+        :param PeerContainer peer_container: The policy will be evaluated against this set of peers
+        :param str file_name: The name of the file in which the istio policy object resides
+        """
+        GenericYamlParser.__init__(self, file_name)
+        self.policy = policy
+        self.peer_container = peer_container
+        self.namespace = None
+        self.referenced_labels = set()
+
+    def parse_generic_istio_policy_fields(self, policy_kind, istio_version):
+        """
+        Parse the common fields in istio policies, e.g kind, apiVersion and metadata
+        :param str policy_kind : the kind of current policy
+        :param str istio_version : the apiVersion of the istio object
+        :return: the name of the current object or None if it is not relevant object
+        :rtype: str
+        """
+        if not isinstance(self.policy, dict):
+            self.syntax_error('type of Top ds is not a map')
+        if self.policy.get('kind') != policy_kind:
+            return None  # Not the relevant object
+        api_version = self.policy.get('apiVersion')
+        if 'istio' not in api_version:
+            return None  # apiVersion is not properly set
+        valid_keys = {'kind': [1, str], 'apiVersion': [1, str], 'metadata': [1, dict], 'spec': [0, dict]}
+        self.check_fields_validity(self.policy, policy_kind, valid_keys,
+                                   {'apiVersion': [istio_version]})
+        metadata = self.policy['metadata']
+        self.check_metadata_validity(metadata)
+        self.namespace = self.peer_container.get_namespace(metadata.get('namespace', 'default'))
```

Comment:
User may get a false warning `Namespace istio-config is missing from the network configuration`

Reply:
In last commit, I updated the get_namespace def to not warn on `istio_root_namespace` since it is a special case.
I also handled new cases regarding this namespace (referring to Noet2 and first example [here](https://istio.io/latest/docs/reference/config/networking/sidecar/)) as following:

- syntax error when global sidecar in root namespace includes workloadSelector.
(on live cluster, behavior of global sidecar with workloadSelector is undefined - as if the sidecar ignored)
- if a host's namespace is `'.'` and sidecar is in the root namespace , this means : allow egress traffic only to other workloads in the same namespace